### PR TITLE
Button/TxButton submit on enter

### LINKED ISSUE #1162: Enter key submits the form
Goal:
Make sure modals and common forms can be submitted with `enter` keys.

Background:
In many cases users that needed to type something as the last parameter of a form, e.g an amount or password, their hand are on the keyboard and they hit the `enter` / `return` key to submit the form to save a couple seconds.

Actions:
- Staking > Bond / BondExtra / Unbond / Set Session 
- Transfer
- Account > Edit / Create / Restore
- Address book > Edit / Create
- Chain state
- Extrinsics
- Toolbox > Unlock

I probably forgot many, but that'd be a start

### PR DESCRIPTION
**Pull request by kwingram25**
Closes #1162 

Add submitOnEnter prop to trigger enter-submit for Button and TxButton and activate on most forms

Needs testing to ensure no event conflicts or side effects, seems safe so far

### CODE REVIEW

File: packages/app-accounts/src/Vanity/index.tsx [hunk 1]
```diff
@@ -98,6 +98,7 @@ class VanityApp extends React.PureComponent<Props, State> {
               ? t('Stop generation')
               : t('Start generation')
           }
+          submitOnEnter={!isRunning}
```
Comment (axelchalon) [inline packages/app-accounts/src/Vanity/index.tsx]: Button doesn't listen to prop changes so I think only the initial value of the prop will be taken into account here

**Comment by kwingram25** [comment]
Testing an approach using refs to trigger buttons/txButtons from Input.onEnter() - currently on Transfer.tsx and Vanity.tsx only

This feels like a lot less elegant in my opinion with a lot more almost-repeating code to add - also needs a lot more work on each possible input component, for example text inputs on InputExtrinsic

**Comment by kwingram25** [comment]
Add TxComponent to link Button/TxButton ref (this.button) with enter events on text inputs (this.submit / this.sendTx)

**Comment by jacogr** [comment]
Some conflicts snuck in.

**Comment by jacogr** [comment]
@Tbaut Since you logged the issue, would like you to play with it as well. (Hence have not merged as-is - it feels fine to me)

**Comment by kwingram25** [comment]
Had to move the ref from class method to constructor, strange bug where it was null on modals

**Pull request merged**